src/Compilers/Core/AnalyzerDriver/DiagnosticDescriptorExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class DiagnosticDescriptorExtensions
+    {
+        public static void SetOnLocalizableStringException(this DiagnosticDescriptor descriptor, DiagnosticAnalyzer owner, Action<Exception, DiagnosticAnalyzer, Diagnostic> onLocalizableStringException)
```

Comment:
Yes, I prefer that too but is it ok to allocate a new descriptor for every descriptor returned by analyzers? Btw the userDiagnosticDescriptor above is already exception safe, we just need to hookup the optional exception handler to report exception diagnostic here.